hydroflow_datalog/src/lib.rs

```diff
@@ -243,23 +248,34 @@ mod tests {
         data
     }
 
-    macro_rules! test_datalog_snapshot {
+    macro_rules! test_snapshots {
         ($program:literal) => {
-            let out = &gen_datalog_program(parse_quote!($program), quote::quote! { hydroflow });
+            let graph = gen_hydroflow_graph(parse_quote!($program));
+
+            insta::with_settings!({snapshot_suffix => "surface_graph"}, {
+                insta::assert_display_snapshot!(graph.surface_syntax_string());
+            });
 
+            // Have to make a new graph as the above closure borrows.
+            let graph2 = gen_hydroflow_graph(parse_quote!($program));
```

Comment:
Can we just clone the graph before it's grabbed by the closure?

Reply:
FlatGraph doesn't implement Clone. I tried Deriving it but Node does not implement Clone.